mobile/test/kotlin/integration/BUILD

```diff
@@ -250,6 +250,20 @@ envoy_mobile_jni_kt_test(
     ],
 )
 
+envoy_mobile_jni_kt_test(
```

Comment:
sorry I had to revert this test becasue example.com doesn't work on CI.  guess I failed to remove the file.
please remove the test and this file and add regression to TEST_P(ProxyFilterIntegrationTest, MultiPortTest) {

Reply:
ProxyFilterIntegrationTest and most other integration tests I observed explicitly specify port in the `Host` header. Whereas the issue this PR is trying to fix happens when the port is implicit derived from the scheme. So other than this test, I couldn't find another one suitable for regression test. Do you have any DNS cache integration test in your mind which doesn't specify a upstream port in request headers?

Reply:
argh, then you need the e2e test which doesn't work on CI for reasons I never tracked down.
let's land this with the unit test because it fixes a real problem, and assign me a follow-up bug to get the e2e tes tworking

Reply:
strike that


+// Regression test a bug where the host header was used for cache lookups rather than host:port key
+TEST_P(ProxyFilterIntegrationTest, CacheSansPort) {
+  useAccessLog("%RESPONSE_CODE_DETAILS%");
+  initializeWithArgs();
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  const Http::TestRequestHeaderMapImpl request_headers{
+      {":method", "POST"},
+      {":path", "/test/long/url"},
+      {":scheme", "http"},
+      {":authority", "localhost"}};
+
+  // Send a request to localhost, with no port specified. The cluster will
+  // default to localhost:443, and the connection will fail.
+  auto response = codec_client_->makeHeaderOnlyRequest(request_headers);
+  ASSERT_TRUE(response->waitForEndStream());
+  EXPECT_EQ("503", response->headers().getStatusValue());
+  EXPECT_THAT(waitForAccessLog(access_log_name_), HasSubstr("upstream_reset_before_response_started"));
+
+  // Now try a second request and make sure it encounters the same error.
+  auto response2 = codec_client_->makeHeaderOnlyRequest(request_headers);
+  ASSERT_TRUE(response2->waitForEndStream());
+  EXPECT_EQ("503", response2->headers().getStatusValue());
+  EXPECT_THAT(waitForAccessLog(access_log_name_, 1), HasSubstr("upstream_reset_before_response_started"));
+}
+
[ RUN      ] IpVersions/ProxyFilterIntegrationTest.CacheSansPort/IPv6
test/extensions/filters/http/dynamic_forward_proxy/proxy_filter_integration_test.cc:513: Failure
Value of: waitForAccessLog(access_log_name_, 1)
Expected: has substring "upstream_reset_before_response_started"
  Actual: "dns_resolution_failure"
Stack trace:
  0x296693e: Envoy::(anonymous namespace)::ProxyFilterIntegrationTest_CacheSansPort_Test::TestBody()
  0x6d6b31b: testing::internal::HandleSehExceptionsInMethodIfSupported<>()
  0x6d5ba1a: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x6d47913: testing::Test::Run()
  0x6d482b4: testing::TestInfo::Run()
... Google Test internal frames ...

Reply:
This is clever! Thanks for the regression test! I can MixedSchemeTest.kt now.